libnavigation-router/src/main/java/com/mapbox/navigation/route/internal/util/DirectionsRouteRefreshUtils.kt

```diff
@@ -0,0 +1,21 @@
+package com.mapbox.navigation.route.internal.util
+
+import com.mapbox.api.directionsrefresh.v1.models.DirectionsRefreshResponse
+import com.mapbox.api.directionsrefresh.v1.models.DirectionsRouteRefresh
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+
+fun parseDirectionsRouteRefresh(
+    json: String,
+): Expected<Throwable, DirectionsRouteRefresh> {
+    return try {
+        val route = DirectionsRefreshResponse.fromJson(json).route()
+        if (route != null) {
+            ExpectedFactory.createValue(route)
+        } else {
+            ExpectedFactory.createError(IllegalStateException("no route refresh returned"))
+        }
+    } catch (ex: NullPointerException) {
```

Comment:
```suggestion
    } catch (ex: Exception) {
```
I feel like being more generic is okay here, we'll deliver the failure in the callback, it's not ignored.